### ISSUE TITLE
zstd and replxx from cmakelists: "no SOURCES given to target"

### ISSUE THREAD

**Issue by modusc896d352** [open-goal/jak-project#2123]
this is my first time opening up jak-project through VS2022.
i have the base thing plus the necessary components ("Desktop development with C++" workload) plus the necessary stuff installed through scoop as listed on README.md, opened CMakeLists.txt through there as said in the Windows build instructions.

and then i get hit with this
```
Gravidade	Código	Descrição	Projeto	Arquivo	Linha	Estado de Supressão
Erro		CMake Error at third-party/zstd/CMakeLists.txt:30 (add_library):
  No SOURCES given to target: libzstd_static		C:\Users\Amon\Desktop\[open-goal][jak-project]\jak-project\third-party/zstd/CMakeLists.txt	30	
Erro		CMake Error at third-party/replxx/CMakeLists.txt:65 (add_library):
  No SOURCES given to target: replxx	replxx	C:\Users\Amon\Desktop\[open-goal][jak-project]\jak-project\third-party/replxx/CMakeLists.txt	65	
```

what this means is that i'm either missing a bunch of files to make it compile or these cmake txt files weren't that well-written to begin with. or maybe i'm just... dumb or something.
either way, jak-project cannot compile on my end.

i simply don't know what to do at this point.

**Comment by ManDude** [comment]
At what point exactly are you getting these errors?

**Comment by modusc896d352** [comment]
let's see...

i open up Visual Studio 2022, and through there i manage to open up a project as a cmake project where i'm supposed to select CMakeLists.txt from the main jak-project folder.

then, after said project is open, i just wait. for a grand total of **3 minutes**. i just let VS2022 check every CMakeLists.txt file out there.

oh but you asked the exact point that these errors just popped off of nowhere. here it is.
```
1> Linha de comando: "C:\WINDOWS\system32\cmd.exe" /c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\Microsoft\CMake\CMake\bin\cmake.exe"  -G "Ninja"  -DCMAKE_C_COMPILER:STRING="clang" -DCMAKE_CXX_COMPILER:STRING="clang++" -DINSTALL_GTEST:STRING="True" -DASAN_BUILD:STRING="OFF" -DSTATICALLY_LINK:STRING="OFF" -DZYDIS_BUILD_SHARED_LIB:STRING="ON" -DCODE_COVERAGE:STRING="OFF" -DCMAKE_BUILD_TYPE:STRING="Debug" -DCMAKE_INSTALL_PREFIX:STRING="C:/Users/Amon/Desktop/[open-goal][jak-project]/jak-project/out/install/Debug-windows-clang"  -DCMAKE_MAKE_PROGRAM="C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\Microsoft\CMake\Ninja\ninja.exe" "C:\Users\Amon\Desktop\[open-goal][jak-project]\jak-project" 2>&1"
1> Diretório de trabalho: C:/Users/Amon/Desktop/[open-goal][jak-project]/jak-project/out/build/Debug
1> [CMake] -- Statically Link? OFF
1> [CMake] -- Clang Detected - Setting Defaults
1> [CMake] Windows SDK version: 7.1.7600.0.30514
1> [CMake] -- Using c++ standard c++17
1> [CMake] -- CMake version: 3.24.202208181-MSVC_2
1> [CMake] -- Project version: 3.2.0
1> [CMake] -- Using Stack Protection hardening
1> [CMake] -- CMAKE_CXX_COMPILER 'C:/Users/Amon/scoop/apps/llvm/current/bin/clang++.exe' 'Clang' '14.0.5'
1> [CMake] -- CMAKE_CXX_FLAGS                '         -fcxx-exceptions         -fexceptions         -fdiagnostics-color=always         -std=c++17         -mavx         -Wall         -Wno-c++11-narrowing         -Wno-c++98-compat         -O3 -Wno-everything  -Wall -Wextra -Wpedantic -Wswitch-enum -Wshadow -Wno-long-long'
1> [CMake] -- CMAKE_CXX_FLAGS_DEBUG          '-g -Xclang -gcodeview -O0 -D_DEBUG -D_DLL -D_MT -Xclang --dependent-lib=msvcrtd'
1> [CMake] -- Compile sqlite3 from source in subdirectory
1> [CMake] -- SQLITECPP_RUN_CPPLINT OFF
1> [CMake] -- SQLITECPP_RUN_CPPCHECK OFF
1> [CMake] -- SQLITECPP_RUN_DOXYGEN OFF
1> [CMake] -- SQLITECPP_BUILD_EXAMPLES OFF
1> [CMake] -- SQLITECPP_BUILD_TESTS OFF
1> [CMake] -- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE) 
1> [CMake] -- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE) 
1> [CMake] -- Including Win32 support
1> [CMake] -- dynamically linking zydis
1> [CMake] -- zydis: dynamically link? ON
1> [CMake] -- Configuring done
1> [CMake] CMake Error at third-party/replxx/CMakeLists.txt:65 (add_library):
1> [CMake]   No SOURCES given to target: replxx
1> [CMake] 
1> [CMake] 
1> [CMake] CMake Error at third-party/zstd/CMakeLists.txt:30 (add_library):
1> [CMake]   No SOURCES given to target: libzstd_static
1> [CMake] 
1> [CMake] 
1> [CMake] CMake Generate step failed.  Build files cannot be regenerated correctly.
1> [CMake] RegularExpression::compile(): Invalid range in [].
1> [CMake] RegularExpression::compile(): Error in compile.
1> [CMake] RegularExpression::compile(): Invalid range in [].
1> [CMake] RegularExpression::compile(): Error in compile.
1> [CMake] RegularExpression::compile(): Invalid range in [].
1> [CMake] RegularExpression::compile(): Error in compile.
1> [CMake] RegularExpression::compile(): Invalid range in [].
1> [CMake] RegularExpression::compile(): Error in compile.
1> [CMake] RegularExpression::compile(): Invalid range in [].
1> [CMake] RegularExpression::compile(): Error in compile.
1> [CMake] RegularExpression::compile(): Invalid range in [].
1> [CMake] RegularExpression::compile(): Error in compile.
1> [CMake] RegularExpression::compile(): Invalid range in [].
1> [CMake] RegularExpression::compile(): Error in compile.
1> [CMake] RegularExpression::compile(): Invalid range in [].
1> [CMake] RegularExpression::compile(): Error in compile.
1> [CMake] RegularExpression::compile(): Invalid range in [].
1> [CMake] RegularExpression::compile(): Error in compile.
1> [CMake] RegularExpression::compile(): Invalid range in [].
1> [CMake] RegularExpression::compile(): Error in compile.
1> [CMake] RegularExpression::compile(): Invalid range in [].
1> [CMake] RegularExpression::compile(): Error in compile.
1> [CMake] RegularExpression::compile(): Invalid range in [].
1> [CMake] RegularExpression::compile(): Error in compile.
1> Falha na execução de 'C:\WINDOWS\system32\cmd.exe' '/c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\Microsoft\CMake\CMake\bin\cmake.exe"  -G "Ninja"  -DCMAKE_C_COMPILER:STRING="clang" -DCMAKE_CXX_COMPILER:STRING="clang++" -DINSTALL_GTEST:STRING="True" -DASAN_BUILD:STRING="OFF" -DSTATICALLY_LINK:STRING="OFF" -DZYDIS_BUILD_SHARED_LIB:STRING="ON" -DCODE_COVERAGE:STRING="OFF" -DCMAKE_BUILD_TYPE:STRING="Debug" -DCMAKE_INSTALL_PREFIX:STRING="C:/Users/Amon/Desktop/[open-goal][jak-project]/jak-project/out/install/Debug-windows-clang"  -DCMAKE_MAKE_PROGRAM="C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\Microsoft\CMake\Ninja\ninja.exe" "C:\Users\Amon\Desktop\[open-goal][jak-project]\jak-project" 2>&1"' com o erro: ''C:\WINDOWS\system32\cmd.exe' '/c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\Microsoft\CMake\CMake\bin\cmake.exe"  -G "Ninja"  -DCMAKE_C_COMPILER:STRING="clang" -DCMAKE_CXX_COMPILER:STRING="clang++" -DINSTALL_GTEST:STRING="True" -DASAN_BUILD:STRING="OFF" -DSTATICALLY_LINK:STRING="OFF" -DZYDIS_BUILD_SHARED_LIB:STRING="ON" -DCODE_COVERAGE:STRING="OFF" -DCMAKE_BUILD_TYPE:STRING="Debug" -DCMAKE_INSTALL_PREFIX:STRING="C:/Users/Amon/Desktop/[open-goal][jak-project]/jak-project/out/install/Debug-windows-clang"  -DCMAKE_MAKE_PROGRAM="C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\Microsoft\CMake\Ninja\ninja.exe" "C:\Users\Amon\Desktop\[open-goal][jak-project]\jak-project" 2>&1"' foi retornado com o código de saída: 1'.
```

**Comment by ManDude** [comment]
Can you try getting rid of the `[` and `]` characters in your path to the project folder?

**Comment by modusc896d352** [comment]
hmmmmmmmmmmmmm
yeah, i can do that but this'll take a while

**Comment by modusc896d352** [comment]
sorry, had to delete the whole repo after said repo has proved useless because VS2022 decided to just mess with it meaning i couldn't do shit with it. then i had to re-clone it.

so i opened the cmake project of that re-cloned repo through VS2022 and this is the result.

```
1> Geração do CMake iniciada para a configuração: 'Debug-windows-clang'.
1> Configurações de ambiente:
1>     CommandPromptType=Native
1>     DevEnvDir=C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\
1>     ExtensionSdkDir=C:\Program Files (x86)\Microsoft SDKs\Windows Kits\10\ExtensionSDKs
1>     EXTERNAL_INCLUDE=C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include;C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\ATLMFC\include;C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Auxiliary\VS\include;C:\Program Files (x86)\Windows Kits\10\include\10.0.20348.0\ucrt;C:\Program Files (x86)\Windows Kits\10\\include\10.0.20348.0\\um;C:\Program Files (x86)\Windows Kits\10\\include\10.0.20348.0\\shared;C:\Program Files (x86)\Windows Kits\10\\include\10.0.20348.0\\winrt;C:\Program Files (x86)\Windows Kits\10\\include\10.0.20348.0\\cppwinrt;C:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um
1>     Framework40Version=v4.0
1>     FrameworkDir=C:\Windows\Microsoft.NET\Framework64\
1>     FrameworkDir64=C:\Windows\Microsoft.NET\Framework64\
1>     FrameworkVersion=v4.0.30319
1>     FrameworkVersion64=v4.0.30319
1>     HTMLHelpDir=C:\Program Files (x86)\HTML Help Workshop
1>     IFCPATH=C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\ifc\x64
1>     INCLUDE=C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include;C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\ATLMFC\include;C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Auxiliary\VS\include;C:\Program Files (x86)\Windows Kits\10\include\10.0.20348.0\ucrt;C:\Program Files (x86)\Windows Kits\10\\include\10.0.20348.0\\um;C:\Program Files (x86)\Windows Kits\10\\include\10.0.20348.0\\shared;C:\Program Files (x86)\Windows Kits\10\\include\10.0.20348.0\\winrt;C:\Program Files (x86)\Windows Kits\10\\include\10.0.20348.0\\cppwinrt;C:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um
1>     LIB=C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\ATLMFC\lib\x64;C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\lib\x64;C:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\lib\um\x64;C:\Program Files (x86)\Windows Kits\10\lib\10.0.20348.0\ucrt\x64;C:\Program Files (x86)\Windows Kits\10\\lib\10.0.20348.0\\um\x64
1>     LIBPATH=C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\ATLMFC\lib\x64;C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\lib\x64;C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\lib\x86\store\references;C:\Program Files (x86)\Windows Kits\10\UnionMetadata\10.0.20348.0;C:\Program Files (x86)\Windows Kits\10\References\10.0.20348.0;C:\Windows\Microsoft.NET\Framework64\v4.0.30319
1>     NETFXSDKDir=C:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\
1>     Path=C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\bin\HostX64\x64;C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\VC\VCPackages;C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\CommonExtensions\Microsoft\TestWindow;C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\CommonExtensions\Microsoft\TeamFoundation\Team Explorer;C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\bin\Roslyn;C:\Program Files\Microsoft Visual Studio\2022\Community\Team Tools\Performance Tools\x64;C:\Program Files\Microsoft Visual Studio\2022\Community\Team Tools\Performance Tools;C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.8 Tools\x64\;C:\Program Files (x86)\HTML Help Workshop;C:\Program Files (x86)\Windows Kits\10\bin\10.0.20348.0\\x64;C:\Program Files (x86)\Windows Kits\10\bin\\x64;C:\Program Files\Microsoft Visual Studio\2022\Community\\MSBuild\Current\Bin\amd64;C:\Windows\Microsoft.NET\Framework64\v4.0.30319;C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\;C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\Tools\;C:\Program Files\Smart Projects\IsoBuster;c:\program files\eclipse adoptium\jdk-17.0.5.8-hotspot\bin;c:\program files\oculus\support\oculus-runtime;c:\program files (x86)\smart projects\isobuster;c:\windows\system32;c:\windows;c:\windows\system32\wbem;c:\windows\system32\windowspowershell\v1.0\;c:\windows\system32\openssh\;c:\program files\dotnet\;c:\program files\winmerge;c:\program files (x86)\nvidia corporation\physx\common;c:\program files\git lfs;c:\program files (x86)\gtksharp\2.12\bin;c:\program files (x86)\gnupg\bin;c:\program files\microsoft\web platform installer\;c:\program files (x86)\microsoft asp.net\asp.net web pages\v1.0\;c:\program files\microsoft sql server\110\tools\binn\;c:\gradle\gradle-7.6\bin;c:\program files\git\cmd;c:\program files\microsoft sql server\150\tools\binn\;c:\program files (x86)\microsoft group policy\windows 10 may 2021 update (21h1)\policydefinitions\;c:\program files\go\bin;c:\program files\git\cmd;c:\program files\010 editor;C:\Program Files\Git\cmd;C:\Users\Amon\scoop\apps\python\current\Scripts;C:\Users\Amon\scoop\apps\python\current;C:\Users\Amon\scoop\apps\llvm\current\bin;C:\Users\Amon\scoop\shims;C:\Users\Amon\.cargo\bin;C:\Users\Amon\AppData\Local\Microsoft\WindowsApps;%PyCharm Edu%;C:\Users\Amon\.dotnet\tools;C:\Users\Amon\AppData\Local\Programs\Microsoft VS Code\bin;C:\Users\Amon\.dotnet\tools;C:\Users\Amon\AppData\Local\Programs\Fiddler;C:\Users\Amon\go\bin;C:\Users\Amon\AppData\Local\ninfs;;C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin;C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\CommonExtensions\Microsoft\CMake\Ninja;C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\VC\Linux\bin\ConnectionManagerExe
1>     PROMPT=$P$G
1>     UCRTVersion=10.0.20348.0
1>     UniversalCRTSdkDir=C:\Program Files (x86)\Windows Kits\10\
1>     VCIDEInstallDir=C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\VC\
1>     VCINSTALLDIR=C:\Program Files\Microsoft Visual Studio\2022\Community\VC\
1>     VCToolsInstallDir=C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\
1>     VCToolsRedistDir=C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Redist\MSVC\14.34.31931\
1>     VCToolsVersion=14.34.31933
1>     VS170COMNTOOLS=C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\Tools\
1>     VSCMD_ARG_app_plat=Desktop
1>     VSCMD_ARG_HOST_ARCH=x64
1>     VSCMD_ARG_no_logo=1
1>     VSCMD_ARG_TGT_ARCH=x64
1>     VSCMD_DEBUG=5 
1>     VSCMD_VER=17.4.4
1>     VSINSTALLDIR=C:\Program Files\Microsoft Visual Studio\2022\Community\
1>     WindowsLibPath=C:\Program Files (x86)\Windows Kits\10\UnionMetadata\10.0.20348.0;C:\Program Files (x86)\Windows Kits\10\References\10.0.20348.0
1>     WindowsSdkBinPath=C:\Program Files (x86)\Windows Kits\10\bin\
1>     WindowsSdkDir=C:\Program Files (x86)\Windows Kits\10\
1>     WindowsSDKLibVersion=10.0.20348.0\
1>     WindowsSdkVerBinPath=C:\Program Files (x86)\Windows Kits\10\bin\10.0.20348.0\
1>     WindowsSDKVersion=10.0.20348.0\
1>     WindowsSDK_ExecutablePath_x64=C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.8 Tools\x64\
1>     WindowsSDK_ExecutablePath_x86=C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.8 Tools\
1>     __DOTNET_ADD_64BIT=1
1>     __DOTNET_PREFERRED_BITNESS=64
1>     __VSCMD_PREINIT_PATH=C:\Program Files\Smart Projects\IsoBuster;c:\program files\eclipse adoptium\jdk-17.0.5.8-hotspot\bin;c:\program files\oculus\support\oculus-runtime;c:\program files (x86)\smart projects\isobuster;c:\windows\system32;c:\windows;c:\windows\system32\wbem;c:\windows\system32\windowspowershell\v1.0\;c:\windows\system32\openssh\;c:\program files\dotnet\;c:\program files\winmerge;c:\program files (x86)\nvidia corporation\physx\common;c:\program files\git lfs;c:\program files (x86)\gtksharp\2.12\bin;c:\program files (x86)\gnupg\bin;c:\program files\microsoft\web platform installer\;c:\program files (x86)\microsoft asp.net\asp.net web pages\v1.0\;c:\program files\microsoft sql server\110\tools\binn\;c:\gradle\gradle-7.6\bin;c:\program files\git\cmd;c:\program files\microsoft sql server\150\tools\binn\;c:\program files (x86)\microsoft group policy\windows 10 may 2021 update (21h1)\policydefinitions\;c:\program files\go\bin;c:\program files\git\cmd;c:\program files\010 editor;C:\Program Files\Git\cmd;C:\Users\Amon\scoop\apps\python\current\Scripts;C:\Users\Amon\scoop\apps\python\current;C:\Users\Amon\scoop\apps\llvm\current\bin;C:\Users\Amon\scoop\shims;C:\Users\Amon\.cargo\bin;C:\Users\Amon\AppData\Local\Microsoft\WindowsApps;%PyCharm Edu%;C:\Users\Amon\.dotnet\tools;C:\Users\Amon\AppData\Local\Programs\Microsoft VS Code\bin;C:\Users\Amon\.dotnet\tools;C:\Users\Amon\AppData\Local\Programs\Fiddler;C:\Users\Amon\go\bin;C:\Users\Amon\AppData\Local\ninfs;
1>     OneDriveConsumer=C:\Users\Amon\OneDrive
1>     ProgramFiles(x86)=C:\Program Files (x86)
1>     ProgramW6432=C:\Program Files
1>     PROCESSOR_IDENTIFIER=AMD64 Family 23 Model 24 Stepping 1, AuthenticAMD
1>     VBOX_MSI_INSTALL_PATH=C:\Program Files\Oracle\VirtualBox\
1>     OnlineServices=Online Services
1>     PROCESSOR_ARCHITECTURE=AMD64
1>     DriverData=C:\Windows\System32\Drivers\DriverData
1>     PkgDefApplicationConfigFile=C:\Users\Amon\AppData\Local\Microsoft\VisualStudio\17.0_4d4ff854\devenv.exe.config
1>     USERPROFILE=C:\Users\Amon
1>     CommonProgramFiles=C:\Program Files\Common Files
1>     GTK_BASEPATH=C:\Program Files (x86)\GtkSharp\2.12\
1>     ServiceHubLogSessionKey=7836CBB0
1>     FPS_BROWSER_APP_PROFILE_STRING=Internet Explorer
1>     FPS_BROWSER_USER_PROFILE_STRING=Default
1>     LOGONSERVER=\\DESKTOP-V9JA6F5
1>     TEMP=C:\Users\Amon\AppData\Local\Temp
1>     USERNAME=Amon
1>     SystemRoot=C:\WINDOWS
1>     VSSKUEDITION=Community
1>     OneDrive=C:\Users\Amon\OneDrive
1>     USERDOMAIN_ROAMINGPROFILE=DESKTOP-V9JA6F5
1>     CommonProgramFiles(x86)=C:\Program Files (x86)\Common Files
1>     VSLANG=1046
1>     ProgramData=C:\ProgramData
1>     platformcode=7F
1>     VSAPPIDDIR=C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\
1>     HOMEPATH=\Users\Amon
1>     COMPUTERNAME=DESKTOP-V9JA6F5
1>     ALLUSERSPROFILE=C:\ProgramData
1>     CommonProgramW6432=C:\Program Files\Common Files
1>     VisualStudioVersion=17.0
1>     ThreadedWaitDialogDpiContext=-4
1>     TMP=C:\Users\Amon\AppData\Local\Temp
1>     VisualStudioDir=C:\Users\Amon\Documents\Visual Studio 2022
1>     SESSIONNAME=Console
1>     PROCESSOR_REVISION=1801
1>     GHIDRA_INSTALL_DIR=C:\Users\Amon\Desktop\ghidra_PUBLIC
1>     windir=C:\WINDOWS
1>     SystemDrive=C:
1>     NUMBER_OF_PROCESSORS=4
1>     OS=Windows_NT
1>     JD2_HOME=C:\Users\Amon\AppData\Local\JDownloader 2.0
1>     ProgramFiles=C:\Program Files
1>     ComSpec=C:\WINDOWS\system32\cmd.exe
1>     VSAPPIDNAME=devenv.exe
1>     HOMEDRIVE=C:
1>     PATHEXT=.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC
1>     GIT_INSTALL_ROOT=C:\Users\Amon\scoop\apps\git\current
1>     JAVA_HOME=C:\Program Files\Eclipse Adoptium\jdk-17.0.5.8-hotspot\
1>     PSModulePath=C:\Program Files\WindowsPowerShell\Modules;C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules
1>     APPDATA=C:\Users\Amon\AppData\Roaming
1>     USERDOMAIN=DESKTOP-V9JA6F5
1>     PROCESSOR_LEVEL=23
1>     LOCALAPPDATA=C:\Users\Amon\AppData\Local
1>     GIT_LFS_PATH=C:\Program Files\Git LFS
1>     VisualStudioEdition=Microsoft Visual Studio Community 2022
1>     RegionCode=LA
1>     DXSDK_DIR=C:\Program Files (x86)\Microsoft DirectX SDK (June 2010)\
1>     PUBLIC=C:\Users\Public
1>     GOPATH=C:\Users\Amon\go
1>     MSBuildLoadMicrosoftTargetsReadOnly=true
1> Linha de comando: "C:\WINDOWS\system32\cmd.exe" /c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "c:\program files\microsoft visual studio\2022\community\common7\ide\commonextensions\microsoft\cmake\CMake\bin\cmake.exe"  -G "Ninja"  -DCMAKE_C_COMPILER:STRING="clang" -DCMAKE_CXX_COMPILER:STRING="clang++" -DINSTALL_GTEST:STRING="True" -DASAN_BUILD:STRING="OFF" -DSTATICALLY_LINK:STRING="OFF" -DZYDIS_BUILD_SHARED_LIB:STRING="ON" -DCODE_COVERAGE:STRING="OFF" -DCMAKE_BUILD_TYPE:STRING="Debug" -DCMAKE_INSTALL_PREFIX:STRING="C:/Users/Amon/Desktop/open-goal_jak-project/jak-project/out/install/Debug-windows-clang"  -DCMAKE_MAKE_PROGRAM="c:\program files\microsoft visual studio\2022\community\common7\ide\commonextensions\microsoft\cmake\Ninja\ninja.exe" "C:\Users\Amon\Desktop\open-goal_jak-project\jak-project" 2>&1"
1> Diretório de trabalho: C:/Users/Amon/Desktop/open-goal_jak-project/jak-project/out/build/Debug
1> [CMake] -- The C compiler identification is Clang 14.0.5 with GNU-like command-line
1> [CMake] -- The CXX compiler identification is Clang 14.0.5 with GNU-like command-line
1> [CMake] -- Detecting C compiler ABI info
1> [CMake] -- Detecting C compiler ABI info - done
1> [CMake] -- Check for working C compiler: C:/Users/Amon/scoop/apps/llvm/current/bin/clang.exe - skipped
1> [CMake] -- Detecting C compile features
1> [CMake] -- Detecting C compile features - done
1> [CMake] -- Detecting CXX compiler ABI info
1> [CMake] -- Detecting CXX compiler ABI info - done
1> [CMake] -- Check for working CXX compiler: C:/Users/Amon/scoop/apps/llvm/current/bin/clang++.exe - skipped
1> [CMake] -- Detecting CXX compile features
1> [CMake] -- Detecting CXX compile features - done
1> [CMake] -- Statically Link? OFF
1> [CMake] -- Clang Detected - Setting Defaults
1> [CMake] Windows SDK version: 7.1.7600.0.30514
1> [CMake] -- Performing Test CMAKE_HAVE_LIBC_PTHREAD
1> [CMake] -- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
1> [CMake] -- Looking for pthread_create in pthreads
1> [CMake] -- Looking for pthread_create in pthreads - not found
1> [CMake] -- Looking for pthread_create in pthread
1> [CMake] -- Looking for pthread_create in pthread - not found
1> [CMake] -- Check if compiler accepts -pthread
1> [CMake] -- Check if compiler accepts -pthread - no
1> [CMake] -- Found Threads: TRUE  
1> [CMake] -- Performing Test COMPILER_HAS_DEPRECATED_ATTR
1> [CMake] -- Performing Test COMPILER_HAS_DEPRECATED_ATTR - Success
1> [CMake] -- Using c++ standard c++17
1> [CMake] -- CMake version: 3.24.202208181-MSVC_2
1> [CMake] -- Project version: 3.2.0
1> [CMake] -- Using Stack Protection hardening
1> [CMake] -- CMAKE_CXX_COMPILER 'C:/Users/Amon/scoop/apps/llvm/current/bin/clang++.exe' 'Clang' '14.0.5'
1> [CMake] -- CMAKE_CXX_FLAGS                '         -fcxx-exceptions         -fexceptions         -fdiagnostics-color=always         -std=c++17         -mavx         -Wall         -Wno-c++11-narrowing         -Wno-c++98-compat         -O3 -Wno-everything  -Wall -Wextra -Wpedantic -Wswitch-enum -Wshadow -Wno-long-long'
1> [CMake] -- CMAKE_CXX_FLAGS_DEBUG          '-g -Xclang -gcodeview -O0 -D_DEBUG -D_DLL -D_MT -Xclang --dependent-lib=msvcrtd'
1> [CMake] -- Compile sqlite3 from source in subdirectory
1> [CMake] -- SQLITECPP_RUN_CPPLINT OFF
1> [CMake] -- SQLITECPP_RUN_CPPCHECK OFF
1> [CMake] -- SQLITECPP_RUN_DOXYGEN OFF
1> [CMake] -- SQLITECPP_BUILD_EXAMPLES OFF
1> [CMake] -- SQLITECPP_BUILD_TESTS OFF
1> [CMake] -- Looking for include file pulse/pulseaudio.h
1> [CMake] -- Looking for include file pulse/pulseaudio.h - not found
1> [CMake] -- Looking for include file alsa/asoundlib.h
1> [CMake] -- Looking for include file alsa/asoundlib.h - not found
1> [CMake] -- Looking for include file jack/jack.h
1> [CMake] -- Looking for include file jack/jack.h - not found
1> [CMake] -- Looking for include file sndio.h
1> [CMake] -- Looking for include file sndio.h - not found
1> [CMake] -- Looking for include file aaudio/AAudio.h
1> [CMake] -- Looking for include file aaudio/AAudio.h - not found
1> [CMake] -- Looking for include file AudioUnit/AudioUnit.h
1> [CMake] -- Looking for include file AudioUnit/AudioUnit.h - not found
1> [CMake] -- Looking for include file audioclient.h
1> [CMake] -- Looking for include file audioclient.h - found
1> [CMake] -- Looking for include files windows.h, mmsystem.h
1> [CMake] -- Looking for include files windows.h, mmsystem.h - found
1> [CMake] -- Looking for include file SLES/OpenSLES.h
1> [CMake] -- Looking for include file SLES/OpenSLES.h - not found
1> [CMake] -- Looking for include file sys/soundcard.h
1> [CMake] -- Looking for include file sys/soundcard.h - not found
1> [CMake] -- Looking for include file android/log.h
1> [CMake] -- Looking for include file android/log.h - not found
1> [CMake] -- Looking for include file sys/audioio.h
1> [CMake] -- Looking for include file sys/audioio.h - not found
1> [CMake] -- Looking for include file kai.h
1> [CMake] -- Looking for include file kai.h - not found
1> [CMake] -- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE) 
1> [CMake] -- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE) 
1> [CMake] -- Including Win32 support
1> [CMake] -- The ASM_NASM compiler identification is NASM
1> [CMake] -- Found assembler: C:/Users/Amon/scoop/shims/nasm.exe
1> [CMake] -- Found Git: C:/Program Files/Git/cmd/git.exe (found version "2.39.0.windows.2") 
1> [CMake] -- Found Python: C:/Users/Amon/scoop/shims/python3.exe (found version "3.10.5") found components: Interpreter 
1> [CMake] no rapidjson, download
1> [CMake] -- dynamically linking zydis
1> [CMake] -- zydis: dynamically link? ON
1> [CMake] -- Configuring done
1> [CMake] -- Generating done
1> [CMake] -- Build files have been written to: C:/Users/Amon/Desktop/open-goal_jak-project/jak-project/out/build/Debug
1> Variáveis do CMake extraídas.
1> Arquivos de origem e cabeçalhos extraídos.
1> Modelo de código extraído.
1> Configurações de cadeia ferramentas extraídas.
1> Extraído inclui caminhos.
1> Geração de CMake concluída.
```

**tl;dr** - VS2022 didn't stop prepping the whole thing until it was finished so no errors prevented it from doing so.

i also uninstalled cmake from my computer but that was while i was a bit hopeless with this whole "zstd" and "replxx" thing so... yeah.

and now, i'll be turning this into "will jak-project compile well on my end?". meaning this isn't over, so expect A LOT of heads up from me in the matter.

**Comment by xTVaser** [comment]
For posterity it's was an underlying cmake issue due to the file path naming of the cloned repository https://discourse.cmake.org/t/regularexpression-compile-invalid-range-in/3707

> and now, i'll be turning this into "will jak-project compile well on my end?". meaning this isn't over, so expect A LOT of heads up from me in the matter.

I hope this doesn't mean our issue page will be flooded with environment specific problems.  The CI runners ensure the project will compile across a variety of platform / compiler combinations.  We're happy to help but please try to do some investigation first (that link was the first result when googling your cmake error)
